armi/reactor/blocks.py

```diff
@@ -2349,29 +2354,31 @@ def autoCreateSpatialGrids(self):
                 )
             )
 
-        ringNumber = hexagon.numRingsToHoldNumCells(self.getNumPins())
-        # For the below to work, there must not be multiple wire or multiple clad types.
-        # note that it's the pointed end of the cell hexes that are up (but the
-        # macro shape of the pins forms a hex with a flat top fitting in the assembly)
+        # build the grid, from pitch and orientation
         grid = grids.HexGrid.fromPitch(
-            self.getPinPitch(cold=True), numRings=0, cornersUp=True
+            self.getPinPitch(cold=True), numRings=0, cornersUp=cornersUp
         )
```

Comment:
Since we are more likely to be calling this on a block w/o a grid, we should "anchor" this grid to the block, e.g.,
`HexGrid.fromPitch(..., armiObject=self)`

This `armiObject` is used to dig up the composite tree on the object attached to the grid

https://github.com/terrapower/armi/blob/d3cc8e5fcb1f83206e372baa657b3d51a7f6496c/armi/reactor/grids/structuredGrid.py#L468-L484

And that is some convoluted logic... I guess it's helpful trying to get global coordinates on e.g., a component in a block so you'd need to dig up to the global coordinate system component -> block -> assembly -> core and find where on the "parent" grid your grid is located?